docs/configuration/ai-analysis.rst

```diff
@@ -299,36 +299,318 @@ Advanced - Customizing HolmesGPT
+            command: "cd {{ repo_dir }} && git log -{{number_of_commits}} --oneline"
+
+          - name: "get_repo_details"
+            description: "Fetches details of a specific repository"
+            command: "curl -H 'Authorization: token {{ github_token }}' https://api.github.com/repos/{{ owner }}/{{ repo }}"
+            parameters:
+              owner:
+                type: "string"
+                description: "Owner of the repository."
+                required: true
+              repo:
+                type: "string"
+                description: "Name of the repository."
+                required: true
+
+          - name: "get_recent_commits"
+            description: "Fetches the most recent commits for a repository"
+            command: "curl -H 'Authorization: token {{ github_token }}' https://api.github.com/repos/{{ owner }}/{{ repo }}/commits?per_page={{ limit }} "
+
+
+**Example 2: Kubernetes Diagnostics Toolset**
+
+This toolset provides diagnostics for Kubernetes clusters, helping developers identify and resolve issues.
+
+
+.. code-block:: yaml
+
+    toolsets:
+      kubernetes/diagnostics:
+        description: "Advanced diagnostics and troubleshooting tools for Kubernetes clusters"
+        docs_url: "https://kubernetes.io/docs/home/"
+        icon_url: "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRPKA-U9m5BxYQDF1O7atMfj9EMMXEoGu4t0Q&s"
+        tags:
+          - core
+        prerequisites:
+          - command: "kubectl version --client"
+        tools:
+          - name: "kubectl_node_health"
+            description: "Check the health status of all nodes in the cluster."
+            command: "kubectl get nodes -o wide"
+          - name: "kubectl_troubleshoot_pod"
+            description: "Fetch logs and describe output for a problematic pod."
+            command: |
+              echo "Logs:" &&
+              kubectl logs {{ pod_name }} -n {{ namespace }} &&
+              echo "\nDescription:" &&
+              kubectl describe pod {{ pod_name }} -n {{ namespace }}
+          - name: "kubectl_check_resource_quota"
+            description: "Fetch the resource quota for a specific namespace."
+            command: "kubectl get resourcequota -n {{ namespace }} -o yaml"
+          - name: "kubectl_pod_disk_usage"
+            description: "Check the disk usage of a specific pod."
+            command: |
+              kubectl exec -n {{ namespace }} {{ pod_name }} -- du -sh /
+          - name: "kubectl_find_evicted_pods"
+            description: "List all evicted pods in a specific namespace."
+            command: "kubectl get pods -n {{ namespace }} --field-selector=status.phase=Failed | grep Evicted"
+          - name: "kubectl_drain_node"
+            description: "Drain a node safely by evicting all pods."
+            command: "kubectl drain {{ node_name }} --ignore-daemonsets --force --delete-emptydir-data"
+
+
+Adding Custom Tools to Holmes
```

Comment:
Is this the same as the two examples above?

Reply:
Basically yes, but previous toolsets were just examples, but here we give a short guide on how to add a toolset

Reply:
Can you give full insttructions on the previous 2 examples as well? (And the exact yaml structure necessary - e.g. `holmes.toolsets` and not just `toolsets`)

So we will have

Example 1 ...
Example 2 ...
Example 3 ...

And each will show the full instructions